### PR TITLE
Utilities: Add 'buggiesay' a cowsay-like program

### PR DESCRIPTION
**Pull request by MacDue**
Inspired by `cowsay` prints an ASCII-art Buggie saying your message, in a line-wrapped speech bubble.
It accepts input from the command-line arguments, as well as standard-in.

ANSI escape sequences are removed with the aid of LibVT, and input is treated as UTF8.

image

I just made this for a bit of fun as `fortune | cowsay` was always my go-to command :)

### CODE REVIEW

**Comment by bgianfo** [comment]
Please wrap the body of your commit message, that's what `Commit linter` is failing. 

```
Utilities: Add 'buggiesay' a cowsay-like program (7654397546ab7c3a1c81440b7a725c7e95e98afa):
  Commit message lines are too long (maximum allowed is 72 characters, except for URLs)
```

**Comment by MacDue** [comment]
Moved the template over to `Base`, and now include somewhat useful `EscapeSequenceStripper` for buffers in LibVT

**Comment by bgianfo** [comment]
@alimpfard Thoughts on the `EscapeSequenceStripper` being added to `LibVT`? Is there a better place?

File: Userland/Utilities/CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ list(APPEND REQUIRED_TARGETS
     touch tr true umount uname uniq uptime w wc which whoami xargs yes less
 )
 list(APPEND RECOMMENDED_TARGETS
-    adjtime aplay abench asctl bt checksum chres cksum copy fortune gunzip gzip init install keymap lsirq lsof lspci man mknod mktemp
```
Comment (sw1tchbl4d3r) [inline Userland/Utilities/CMakeLists.txt:10]: Why is `install` being removed here?

Comment (MacDue) [reply Userland/Utilities/CMakeLists.txt:10]: Mistake in conflict resolution (fixed now)

File: Userland/Utilities/buggiesay.cpp [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    struct BoxEdge {
+        char left, right;
+    };
+
+    auto get_box_edge = [&](size_t line_num) {
+        if (line_num == 0)
+            return BoxEdge { '/', '\\' };
+        else if (line_num == lines.size() - 1)
+            return BoxEdge { '\\', '/' };
+        else
```
Comment (alimpfard) [inline Userland/Utilities/buggiesay.cpp:166]: same here.

**Review by alimpfard — COMMENTED** [review]
(Sorry for the very late review, this probably got lost in the hundreds of notifications I have)

@bgianfo I think LibVT is pretty much the best place to put that, as it knows how it will be parsing escape sequences.

File: Userland/Libraries/LibVT/EscapeSequenceStripper.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+private:
+    inline void emit_code_point(u32 code_point) override
+    {
+        (void)AK::UnicodeUtils::code_point_to_utf8(code_point, [&](char utf8) {
+            rewrite_byte(utf8);
+        });
+    }
+    inline void execute_control_code(u8 code) override
+    {
+        if (code == '\t' || code == '\n')
```
[the remark this answers is not in the document]
Comment (MacDue) [reply Userland/Libraries/LibVT/EscapeSequenceStripper.h]: I've made `allowed_control_characters` an optional parameter of the `EscapeSequenceStripper`. If not set it allows all control characters. For `buggiesay` though we only want `\t` or `\n` (other control characters could break the formatting).

**Comment by MacDue** [comment]
I think the changes have been addressed, and I'm not ready for the grim spectre of stale bot to take my `buggiesay` :pensive:

**Comment by Diogenes31** [comment]
I think this utility fits with the theme of the greater project. It's very lighthearted and it *is* a classic unix game.

**Comment by awesomekling** [comment]
Let's not add "funny" programs to the base system. If you want to add things like this, please do it as a port :)

**Pull request closed without merging**